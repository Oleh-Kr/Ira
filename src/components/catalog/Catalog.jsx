import React from "react";
import axios from "axios";
import { Link, useNavigate } from "react-router-dom";
import styles from "./Catalog.module.scss"

const baseURL = "https://api.spoonacular.com/recipes/findByNutrients?minCarbs=45&maxCarbs=50&number=30&apiKey=60c1ca27b27b4afbb847decdc9d34f04";

export default function Catalog() {

  const [post, setPost] = React.useState(null);
  const navigate = useNavigate();

  React.useEffect(() => {
    axios.get(baseURL).then((response) => {
      console.log(response.data)
      setPost(response.data);
    });
  }, []);

  if (!post) return null;

  function handleCardClick(e){
    e.preventDefault()
    const id = e.target.dataset.id
    localStorage.setItem("card-id", id)
    navigate("/product")
  }
  return (
    <>
    <ul className={styles.catalogSection}>
        {post.map((data, i) =>(
        <li key={data.image} className={styles.catalogLi} onClick={handleCardClick} data-id={data.id}>
        <Link to="/product" className={styles.catalogLink}><img src={data.image} alt="cupcake" /></Link>
        <p>{data.title}</p>
        </li>
        ))}
    
    </ul>
    </>
  );
 
}
