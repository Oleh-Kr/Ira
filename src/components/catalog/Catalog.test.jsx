import React from "react";
import { BrowserRouter as Router, Link, useNavigate } from "react-router-dom";
import renderer from "react-test-renderer";
import Catalog from "./Catalog";
import axios from "axios";

jest.mock("axios");

describe("Catalog Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly when API call is successful", () => {
    const mockData = [
      { id: 1, image: "image1.jpg", title: "Product 1" },
      { id: 2, image: "image2.jpg", title: "Product 2" },
    ];

    axios.get.mockResolvedValueOnce({ data: mockData });

    const component = renderer.create(
      <Router>
        <Catalog />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders nothing when the API call fails", () => {
    axios.get.mockRejectedValueOnce(new Error("API Error"));

    const component = renderer.create(
      <Router>
        <Catalog />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toBeNull();
  });

  test("navigates to the product page on card click", () => {
    const mockData = [
      { id: 1, image: "image1.jpg", title: "Product 1" },
      { id: 2, image: "image2.jpg", title: "Product 2" },
    ];

    axios.get.mockResolvedValueOnce({ data: mockData });

    const navigate = jest.fn();
    jest.spyOn(useNavigate, "default").mockReturnValue(navigate);

    const component = renderer.create(
      <Router>
        <Catalog />
      </Router>
    );
    const card = component.root.findByProps({ "data-id": 1 });

    card.props.onClick({ preventDefault: jest.fn() });

    expect(navigate).toHaveBeenCalledWith("/product");
    expect(localStorage.setItem).toHaveBeenCalledWith("card-id", 1);
  });
});
