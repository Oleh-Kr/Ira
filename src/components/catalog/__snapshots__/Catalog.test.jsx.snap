// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog Component renders correctly when API call is successful 1`] = `null`;
